### PR TITLE
Fixed related issues- featureProcedure

### PR DESCRIPTION
**Pull request by StefaniaBeldean24**
I created a PR from my local 'featureProcedure' branch to the remote 'featureProcedure' branch

### CODE REVIEW

File: src/main/java/com/internship/backend/config/UsernamePasswordAuthenticationProvider.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class UsernamePasswordAuthenticationProvider implements AuthenticationPro
     public Authentication authenticate(Authentication authentication) throws AuthenticationException {
         String username = authentication.getName();
         String password = authentication.getCredentials().toString();
-        Optional<User> optionalUser = userRepository.findByUsername(username);
+        Optional<User> optionalUser = Optional.ofNullable(userRepository.findByUsername(username));
```
Comment (solcanm) [inline src/main/java/com/internship/backend/config/UsernamePasswordAuthenticationProvider.java:33]: I think that we can use type inference here: https://docs.oracle.com/en/java/javase/17/language/local-variable-type-inference.html

Also, we can use static import for ofNullable(...) method

File: src/main/java/com/internship/backend/controller/UserController.java [hunk 2]
```diff
@@ -55,8 +51,14 @@ public Optional<User> getUserById(@PathVariable Integer id) {
     }
 
     @PutMapping("/update/{id}")
-    public User updateUser(@PathVariable Integer id, @RequestBody User user) {
-        return userService.updateUser(id, user);
+    public ResponseEntity<Object> updateUser(@PathVariable Integer id, @RequestBody UserDTO userDTO) {
+        try{
+            User user = userService.parse(userDTO);
```
Comment (solcanm) [inline src/main/java/com/internship/backend/controller/UserController.java:56]: When a variable is used just once, inlining it helps improve readability.

File: src/main/java/com/internship/backend/dto/LocationDTO.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
 @NoArgsConstructor
 @AllArgsConstructor
 public class LocationDTO {
-    private int id;
+    //private int id;
```
Comment (solcanm) [inline src/main/java/com/internship/backend/dto/LocationDTO.java:14]: Should the id not being defined at all?

File: src/main/java/com/internship/backend/service/ReservationService.java [hunk 4]
```diff
@@ -40,6 +40,24 @@ public List<Reservation> getAllReservations(){
...
+
+        addReservationValidation(reservation);
+
+        if (isValidReservation((reservation)))
+        {
+            return reservationRepository.save(reservation);
+        }
+        else{
+            throw new ReservationAlreadyExists("Reservation intersects another one");
+        }
```
Comment (solcanm) [inline src/main/java/com/internship/backend/service/ReservationService.java:52]: Indent this lines. 
https://www.javatpoint.com/java-indentation

File: src/test/java/com/internship/backend/service/ReservationServiceTest.java [hunk 5]
```diff
@@ -1,12 +1,168 @@
...
+        //act
+        Reservation result = reservationService.addReservation(reservation);
+
+        //assert
+        assertNotNull(result);
+        assertEquals(reservation, result);
+    }
+
+    @Test
+    void addReservation_invalidReservation() throws InvalidDateException, ReservationAlreadyExists {
```
Comment (solcanm) [inline src/test/java/com/internship/backend/service/ReservationServiceTest.java:69]: This test could be renamed to something like: shouldNotSaveInvalidReservations ?

File: src/main/java/com/internship/backend/controller/UserController.java [hunk 6]
```diff
@@ -55,8 +51,14 @@ public Optional<User> getUserById(@PathVariable Integer id) {
     }
 
     @PutMapping("/update/{id}")
-    public User updateUser(@PathVariable Integer id, @RequestBody User user) {
-        return userService.updateUser(id, user);
+    public ResponseEntity<Object> updateUser(@PathVariable Integer id, @RequestBody UserDTO userDTO) {
+        try{
```
Comment (breazc) [inline src/main/java/com/internship/backend/controller/UserController.java:55]: missing space

File: src/main/java/com/internship/backend/model/Authority.java [hunk 7]
```diff
@@ -20,6 +23,11 @@ public class Authority {
 
     @JsonBackReference
     @ManyToOne
-    @JoinColumn(name = "users_id")
-    private Users user;
+    @JoinColumn(name = "user_id")
+    private User user;
+
+
```
Comment (breazc) [inline src/main/java/com/internship/backend/model/Authority.java:29]: remove space

File: src/test/java/com/internship/backend/config/UsernamePasswordAuthenticationProviderTest.java [hunk 8]
```diff
@@ -44,10 +43,10 @@ void authenticate_ValidCredentials() {
 
         User user = new User();
```
Comment (breazc) [inline src/test/java/com/internship/backend/config/UsernamePasswordAuthenticationProviderTest.java:44]: could you extract this part in a separate method and use it in the rest of the tests?

File: src/test/java/com/internship/backend/service/ReservationServiceTest.java [hunk 9]
```diff
@@ -1,12 +1,168 @@
...
+        List<Reservation> result = reservationService.getAllReservations();
+
+        //assert
+        assertEquals(result, reservations);
+        verify(reservationRepository, times(1)).findAll();
+    }
+
     @Test
-    void getAvailableTennisCourts() {
+    void update_Valid() throws ReservationAlreadyExists {
```
Comment (breazc) [inline src/test/java/com/internship/backend/service/ReservationServiceTest.java:103]: you could rename the tests more suggestively. for example this test can be renamed shouldUpdateWhenReservationIsValid or something

File: src/test/java/com/internship/backend/service/ReservationServiceTest.java [hunk 10]
```diff
@@ -1,12 +1,168 @@
...
+
+    @Test
+    void delete_Valid() throws ReservationDoesNotExistException {
+        when(reservationRepository.existsById(1)).thenReturn(true);
+        when(reservationRepository.count()).thenReturn(0L);
+
+        reservationService.delete(1);
+
+        verify(reservationRepository, times(1)).deleteById(1);
+        verify(priceRepository, times(1)).deleteById(1);
```
Comment (breazc) [inline src/test/java/com/internship/backend/service/ReservationServiceTest.java:134]: times(1) is default, can be deleted

**Pull request merged**